apps/src/lib/ui/SchoolInfoConfirmationDialog.jsx

```diff
@@ -15,80 +15,96 @@ export const styles = {
 class SchoolInfoConfirmationDialog extends Component {
   static propTypes = {
     schoolName: PropTypes.string,
-    onUpdate: PropTypes.func,
-    onConfirm: PropTypes.func,
-    isOpen: PropTypes.bool.isRequired
+    scriptData: PropTypes.shape({
+      formUrl: PropTypes.string.isRequired,
+      authTokenName: PropTypes.string.isRequired,
+      authTokenValue: PropTypes.string.isRequired,
+      existingSchoolInfo: PropTypes.shape({
+        id: PropTypes.number,
+        school_id: PropTypes.string,
+        country: PropTypes.string,
+        school_type: PropTypes.string,
+        school_name: PropTypes.string,
+        full_address: PropTypes.string
+      }).isRequired
+    }).isRequired,
+    onClose: PropTypes.func,
+    isOpen: PropTypes.bool
   };
 
   constructor(props) {
     super(props);
     this.state = {
       showSchoolInterstitial: false,
-      schoolName: props.schoolName,
-      isOpen: true
+      schoolName: props.scriptData.existingSchoolInfo.school_name,
+      isOpen: props.isOpen || true
     };
   }
 
-  static defaultProps = {
-    onUpdate: () => {},
-    schoolName: ''
-  };
+  handleClickYes = () => {
+    const {authTokenName, authTokenValue} = this.props.scriptData;
 
-  componentDidMount() {
-    const {schoolName} = this.state;
-    if (!schoolName && schoolName.length > 0) {
-      fetch('/dashboardapi/v1/users/me/school_name')
-        .then(response => response.json())
-        .then(data => {
-          this.setState({
-            schoolName: data.school_name
-          });
-        })
-        .catch(error => this.setState({error}));
-    }
-  }
+    console.log(authTokenName, authTokenValue);
+
+    fetch(
+      `/api/v1/user_school_infos/${
+        this.props.scriptData.existingSchoolInfo.id
+      }/update_last_confirmation_date`,
+      {method: 'PATCH', headers: {[authTokenName]: authTokenValue}}
+    )
+      .then(() => this.props.onClose())
+      .catch(error => {
+        console.log('The Error =>', error);
+        this.setState({error});
+      });
+  };
 
-  handleUpdateClick = () => {
-    this.setState({showSchoolInterstitial: true});
+  handleClickUpdate = () => {
+    console.log('===>I have been called');
+    const {authTokenName, authTokenValue} = this.props.scriptData;
+    fetch(
+      `/api/v1/user_school_infos/${
+        this.props.scriptData.existingSchoolInfo.id
+      }/update_end_date`,
```

Comment:
Does it make sense to update the end date as soon as the teacher clicks "Update" here, or not until they provide us with new school info on the next screen?  Right now I worry about the experience of clicking "Update" and then close out of the school info interstitial.  We'll record that they are no longer at their current school, but not have new school info for them.  In the meantime, I think the teacher's expectation would be that nothing permanent had occurred - they cancelled out, after all.

On a related note it looks like the new `update_school_info_id` route is still unused.  Is that a next step?

Reply:
@islemaster The update_school_info_id is in use (There is code that was merged into this branch yet, glad one of the tests caught it).

Reply:
@islemaster Good points you raised.  
The current flow is that when the user clicks on the update button, the following fields are updated:  the “end_date” in the user_school_infos, last_seen_school_info_interstitial in the properties column of the user’s table and the last_confirmation_date in the users table.

The school info confirmation modal is closed before the school info interstitial form is rendered.
(the handleClickSave function was missing)

 handleClickSave = () => {
    const {authTokenName, authTokenValue} = this.props.scriptData;
    fetch(
      `/api/v1/users/${
        this.props.scriptData.existingSchoolInfo.id
      }/update_school_info_id`,
      {method: 'PATCH', headers: {[authTokenName]: authTokenValue}}
    )
      .then(() => this.props.onClose)
      .catch(() => {});
  };

There is a check that is made when a teacher clicks the "save" button.  If the school info interstitial form is not completely filled out, the teacher will continue to see the school interstitial form each time the teacher signs in.